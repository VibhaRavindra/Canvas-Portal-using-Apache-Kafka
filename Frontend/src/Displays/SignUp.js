import React, { Component } from 'react';
import '../App.css';
import { Redirect,Link } from 'react-router-dom';
import { signUp } from "../js/actions/actions";
import { connect } from "react-redux";

function mapStateToProps(state) {
    return {
        isLoggedIn:state.isLoggedIn,
        signedUpFail:state.signedUpFail,
        signUpFailMessage:state.signUpFailMessage
    }
}

function mapDispatchToProps(dispatch) {
    return {
        signUp: (data) => dispatch(signUp(data))
    };
}

class SignUp extends Component {

    constructor(props){
        super(props);
        this.submitSignUp = this.submitSignUp.bind(this);
    }
    submitSignUp(e) {
        e.preventDefault();
        const data = new FormData(e.target);
        this.props.signUp(data);
    }
    render(){
        let failCase;
        console.log(this.props.isLoggedIn)
        if(this.props.isLoggedIn){
            return(<Redirect to='/dashboard' />);
        } else if (this.props.signedUpFail!= null && this.props.signedUpFail === false) {
            return(<Redirect to='/signin' />);
        } else if(this.props.signedUpFail === true){
            failCase = <div className="error-box">
            <span>
            <img className="error-icon" src="https://www.freeiconspng.com/uploads/alert-icon-red-11.png" alt="" />
            </span>
            <span className="error-msg">{this.props.signUpFailMessage}</span>
        </div>
        }
        return(
            <div>
            <div className="LoginPage">
                <div className="imageHeader">
                    <img className="sjsuImage" src="https://ok2static.oktacdn.com/bc/image/fileStoreRecord?id=fs01heub3azJBMXWF0x7" alt="" />
                    <div className="beacon-container"></div>
                </div>
                
                <div className="LogIncontainer">
                    <p className="signInText">Sign Up</p>
                    {failCase} 
                    <form className="form"  onSubmit={this.submitSignUp}>
                    <input type="text" className="inputStyle1" name="username" placeholder="Name" required />  
                    <input type="email" className="inputStyle1" name="emailId" placeholder="SJSU Email ID" required />                
                    <input type="password" className="inputStyle1" name="password" placeholder="Password" required /> <br />
                    Student<input type="radio" value="student" name="employee" checked/>  
                    Faculty<input type="radio" value="faculty" name="employee" /> 
                    <button type="submit" className="signInBtn">Sign Up</button>
                    </form>
                    <Link to="/signin"><button className="orText">Already signed up?</button></Link>
                </div> 
            </div>
            </div>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);