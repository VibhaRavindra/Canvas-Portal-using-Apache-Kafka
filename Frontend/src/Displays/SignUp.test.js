import React from 'react';
import SignUp from './SignUp';
import renderer from 'react-test-renderer';
import { mount, shallow, render } from 'enzyme';
import { Provider } from "react-redux";
import store from "../js/store/store";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from "react-router-dom";
configure({ adapter: new Adapter() });

it('Signup renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Router><SignUp/></Router></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });