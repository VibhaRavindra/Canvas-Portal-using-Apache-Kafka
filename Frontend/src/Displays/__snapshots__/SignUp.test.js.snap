// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders correctly 1`] = `
<div>
  <div
    className="LoginPage"
  >
    <div
      className="imageHeader"
    >
      <img
        alt=""
        className="sjsuImage"
        src="https://ok2static.oktacdn.com/bc/image/fileStoreRecord?id=fs01heub3azJBMXWF0x7"
      />
      <div
        className="beacon-container"
      />
    </div>
    <div
      className="LogIncontainer"
    >
      <p
        className="signInText"
      >
        Sign Up
      </p>
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="inputStyle1"
          name="username"
          placeholder="Name"
          required={true}
          type="text"
        />
        <input
          className="inputStyle1"
          name="emailId"
          placeholder="SJSU Email ID"
          required={true}
          type="email"
        />
        <input
          className="inputStyle1"
          name="password"
          placeholder="Password"
          required={true}
          type="password"
        />
         
        <br />
        Student
        <input
          checked={true}
          name="employee"
          type="radio"
          value="student"
        />
        Faculty
        <input
          name="employee"
          type="radio"
          value="faculty"
        />
        <button
          className="signInBtn"
          type="submit"
        >
          Sign Up
        </button>
      </form>
      <a
        href="/signin"
        onClick={[Function]}
      >
        <button
          className="orText"
        >
          Already signed up?
        </button>
      </a>
    </div>
  </div>
</div>
`;
